test/unit-test/FreeRTOS_BitConfig/FreeRTOS_BitConfig_utest.c

```diff
@@ -0,0 +1,617 @@
+    xConfig.uxSize = SIZE_OF_BINARY_STREAM;
+
+    vBitConfig_write_uc( &xConfig, pucData, SIZE_OF_BINARY_STREAM );
+
+    TEST_ASSERT_EQUAL( pdTRUE, xConfig.xHasError );
+}
+
+/**
+ * @brief This functions verifies writing SIZE_OF_BINARY_STREAM
+ *        bytes from the bit stream.
+ */
+
+void test_vBitConfig_write_uc_HappyPath( void )
+{
+    BitConfig_t xConfig;
+    size_t uxSize = SIZE_OF_BINARY_STREAM;
+    uint8_t ucContents[ uxSize ];
+    uint8_t ucData[ uxSize ];
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( ucContents, 0, uxSize );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_uc( &xConfig, ucData, uxSize );
+
+    TEST_ASSERT_EQUAL( SIZE_OF_BINARY_STREAM, xConfig.uxIndex );
+}
+
+/**
+ * @brief This functions verifies writing a
+ *        byte to the bit stream
+ */
+void test_vBitConfig_write_8( void )
+{
+    BitConfig_t xConfig;
+    size_t uxSize = SIZE_OF_BINARY_STREAM;
+    uint8_t ucValue = 0;
+    uint8_t ucContents[ uxSize ];
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( ucContents, 0, uxSize );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_8( &xConfig, ucValue );
+
+    TEST_ASSERT_EQUAL( sizeof( uint8_t ), xConfig.uxIndex );
+}
+
+/**
+ * @brief This functions verifies writing a
+ *        2 byte to the bit stream
+ */
+void test_vBitConfig_write_16( void )
+{
+    BitConfig_t xConfig;
+    size_t uxSize = SIZE_OF_BINARY_STREAM;
+    uint16_t usValue = 0;
+    uint8_t ucContents[ uxSize ];
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( ucContents, 0, uxSize );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_16( &xConfig, usValue );
+
+    TEST_ASSERT_EQUAL( sizeof( uint16_t ), xConfig.uxIndex );
+}
+
+/**
+ * @brief This functions verifies writing a
+ *        4 byte to the bit stream
+ */
+void test_vBitConfig_write_32( void )
+{
+    BitConfig_t xConfig;
+    size_t uxSize = SIZE_OF_BINARY_STREAM;
+    uint32_t usValue = 0;
+    uint8_t ucContents[ uxSize ];
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    xConfig.xHasError = pdFALSE;
+    xConfig.uxIndex = 0;
+    xConfig.uxSize = uxSize;
+    xConfig.ucContents = ucContents;
+
+    vBitConfig_write_32( &xConfig, usValue );
+
+    TEST_ASSERT_EQUAL( sizeof( uint32_t ), xConfig.uxIndex );
+}
+
+/**
+ * @brief This functions verifies failure in
+ *        release the buffer as it is NULL.
+ */
+void test_vBitConfig_releaseNULL( void )
+{
+    BitConfig_t xConfig, xConfigExpected;
+
+    memset( &xConfig, 0, sizeof( BitConfig_t ) );
+    memset( &xConfigExpected, 0, sizeof( BitConfig_t ) );
+    xConfig.ucContents = NULL;
+
+    vBitConfig_release( &xConfig );
```

Comment:
Can we add a test case for NULL pointer like below?
`vBitConfig_release( NULL );`